src/components/AgreementSections/InfoPeriods.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  Button,
+  ButtonGroup,
+  Col,
+  FormattedUTCDate,
+  Headline,
+  KeyValue,
+  Layout,
+  NoValue,
+  Row,
+} from '@folio/stripes/components';
+
+export default function InfoPeriods({ periods }) {
+  /*
+   * This could be handled by callback, but since we need to know for display whether next/prev/current exist
+   * it seems more prudent to just fetch all three at once and then do our logic at this stage
+   */
+  const displayPeriodOptions = {
+    currentPeriod: periods?.find(p => p.periodStatus === 'current'),
+    nextPeriod: periods?.find(p => p.periodStatus === 'next'),
+    previousPeriod: periods?.find(p => p.periodStatus === 'previous')
+  };
+
+  /*
+   * Set up which period is currently selected. Heirachy is Current > Next > Previous
+   * The empty array as a second argument to useEffect should mean this code only runs once to prevent infinite loops
+   */
+  const [selectedPeriod, setSelectedPeriod] = useState('current');
+  useEffect(() => {
+    if (displayPeriodOptions.currentPeriod === undefined) {
+      if (displayPeriodOptions.nextPeriod !== undefined) {
+        setSelectedPeriod('next');
+      } else {
+        setSelectedPeriod('previous');
```

Comment:
Can we avoid working with `undefined`s? and instead default the values generated in displayPeriodOptions to maybe `{}` and work with them assuming each individual period there is an object?

Reply:
So use something like Object.keys({}).isEmpty to check for blank..? I can do but isn't that just harder to understand. Why not work with undefined?

Reply:
Change made anyway :)